examples/02_decoding/plot_miyawaki_encoding.py

```diff
@@ -0,0 +1,233 @@
+# Building the encoding models
+# ----------------------------
+# We can now proceed to do a simple encoding using Ridge regression.
+
+from sklearn.linear_model import Ridge
+from sklearn.cross_validation import KFold
+
+##############################################################################
+# Using 10-fold cross-validation, we partition the data into 10 'Folds',
+# we hold out each fold of the data for testing, then fit a ridge regression
+# to the remaining 9/10 of the data, using y_train as predictors
+# and X_train as targets, and create predictions for the held-out 10th.
+
+estimator = Ridge(alpha=100.)
+cv = KFold(len(y_train), 10)
+
+predictions = [Ridge(alpha=100.).fit(
+    y_train.reshape(-1, 100)[train], X_train[train]).predict(
+        y_train.reshape(-1, 100)[test])
+    for train, test in cv]
+
+
+##############################################################################
+# To have a measure for the quality of our encoding model, we estimate how
+# much variance our encoding model explains in each voxel.
+
+scores = [1. - (((X_train[test] - pred) ** 2).sum(axis=0) /
+         ((X_train[test] - X_train[test].mean(axis=0)) ** 2).sum(axis=0))
+         for pred, (train, test) in zip(predictions, cv)]
+
+##############################################################################
+# Mapping the encoding scores on the brain
+# ----------------------------------------
+# To plot the scores onto the brain, we use :func:`nilearn.image.new_img_like`
+# to create a Nifti1Image containing the scores and then threshold it:
+
+from nilearn.image import threshold_img, new_img_like
+from nilearn.image.resampling import coord_transform
+
+cut_score = np.array(scores).mean(axis=0)
+cut_score[cut_score < 0] = 0
+
+# bring the scores into the shape of the background brain
+score_map = new_img_like(dataset.background,
+                         masking.unmask(cut_score, dataset.mask).get_data())
+
+thresholded_score_map = threshold_img(score_map, threshold = 1e-6)
+
+##############################################################################
+# Plotting the statistical map on a background brain, we mark four voxels
+# which we will inspect more closely later on.
+
+def index_to_xy_coord(x,y,z=10):
+    '''Transforms data index to coordinates of the background + offset'''
+    coords = coord_transform(x,y,z,affine=thresholded_score_map.get_affine())
+    return coords + np.array([0,1,0])
+
+
+xy_indices_of_special_voxels = [(30, 10), (32, 10), (31, 9), (31, 10)]
+
+statmap = plot_stat_map(thresholded_score_map, bg_img=dataset.background,
+                        cut_coords=[-8], display_mode='z', aspect=1.25,
+                        title='Explained variance per voxel',
+                        symmetric_cbar=False)
+
+# creating a contour for each voxel and adding it to the statistical map
+
+for i, (x, y) in enumerate(xy_indices_of_special_voxels):
+    statmap.add_markers(np.array(index_to_xy_coord(x,y))[np.newaxis,:],
```

Comment:
Can you modify index_to_xy_coord so that it's return argument can be passed to add_markers with any transformation. It will make the example simpler.